.github/workflows/build-and-push-image.yml

```diff
@@ -58,7 +58,7 @@ jobs:
         uses: docker/build-push-action@v3
         with:
           context: .
-          file: Dockerfile
+          file: docker/Dockerfile
```

Comment:
We should use `working-directory: docker` at the task level instead of changing the path so then there's no need to change the docker compose file too.